addons/knative/enable

```diff
@@ -4,30 +4,111 @@ set -e
-                  "https://github.com/knative/serving/releases/download/v0.24.0/serving-default-domain.yaml"
-                  "https://github.com/knative/eventing/releases/download/v0.24.0/eventing-core.yaml"
-                  "https://github.com/knative/eventing/releases/download/v0.24.0/in-memory-channel.yaml"
-                  "https://github.com/knative/eventing/releases/download/v0.24.0/mt-channel-broker.yaml"
-                 )
-
-for yaml in "${yamls[@]}"
+for i in "$@"
 do
-   $KUBECTL apply -f "$yaml"
-   sleep 3
+case $i in
+    --no-serving)
+    KNATIVE_SERVING=false
+    shift # past argument
+    ;;
+    --eventing)
+    KNATIVE_EVENTING=true
+    shift # past argument
+    ;;
+    *)
+          # unknown option
+    ;;
+esac
 done
 
+#requirement
+"$SNAP/microk8s-enable.wrapper" dns
+
+echo "Enabling Knative ${KNATIVE_VERSION}"
+$KUBECTL apply -f https://github.com/knative/operator/releases/download/knative-v$KNATIVE_VERSION/operator.yaml
+
+if [ "$KNATIVE_SERVING" = "true" ]; then
+    echo "Installing Knative Serving ${KNATIVE_VERSION}"
+$KUBECTL apply -f- << EOF
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: knative-serving
+---
+apiVersion: operator.knative.dev/v1beta1
+kind: KnativeServing
+metadata:
+  name: knative-serving
+  namespace: knative-serving
+spec:
+  ingress:
+    kourier:
+      enabled: true
+  config:
+    network:
+      ingress-class: "kourier.ingress.networking.knative.dev"
+EOF
+fi
+
+if [ "$KNATIVE_EVENTING" = "true" ]; then
+   echo "Installing Knative Eventing ${KNATIVE_VERSION}"
+$KUBECTL apply -f- << EOF
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: knative-eventing
+---
+apiVersion: operator.knative.dev/v1beta1
+kind: KnativeEventing
+metadata:
+  name: knative-eventing
+  namespace: knative-eventing
+EOF
+fi
+
+# Handle Knative CLIs
+if [ ! -f "${SNAP_DATA}/bin/kn" ]; then
+  echo "Fetching Knative CLI version ${KNATIVE_VERSION} in $SNAP_DATA/bin/kn"
+  run_with_sudo mkdir -p "$SNAP_DATA/bin"
+  fetch_as https://github.com/knative/client/releases/download/knative-v${KNATIVE_VERSION}/kn-linux-${ARCH} "$SNAP_DATA/bin/kn"
+  run_with_sudo chmod uo+x "$SNAP_DATA/bin/kn"
+fi
+if [ ! -f "${SNAP_DATA}/bin/kn-admin" ]; then
+  echo "Fetching Knative CLI Admin plugin version ${KNATIVE_VERSION} in $SNAP_DATA/bin/kn-admin"
+  run_with_sudo mkdir -p "$SNAP_DATA/bin"
+  fetch_as https://github.com/knative-sandbox/kn-plugin-admin/releases/download/knative-v${KNATIVE_VERSION}/kn-admin-linux-${ARCH} "$SNAP_DATA/bin/kn-admin"
+  run_with_sudo chmod uo+x "$SNAP_DATA/bin/kn-admin"
+fi
+if [ ! -f "${SNAP_DATA}/bin/kn-event" ]; then
+  echo "Fetching Knative CLI Event plugin version ${KNATIVE_VERSION} in $SNAP_DATA/bin/kn-event"
+  run_with_sudo mkdir -p "$SNAP_DATA/bin"
+  fetch_as https://github.com/knative-sandbox/kn-plugin-event/releases/download/knative-v${KNATIVE_VERSION}/kn-event-linux-${ARCH} "$SNAP_DATA/bin/kn-event"
```

Comment:
The url with `${ARCH}` (x86_64, aarch64) returns a 404. Use `${SNAP_ARCH}` instead (which is `amd64`, `arm64`). Same for the other downloads. 

```suggestion
  fetch_as https://github.com/knative-sandbox/kn-plugin-event/releases/download/knative-v${KNATIVE_VERSION}/kn-event-linux-${SNAP_ARCH} "$SNAP_DATA/bin/kn-event"
```